### ISSUE TITLE
[HOLD #50349] Chat - Opened offline image preview switches to previously uploaded image online

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#50296]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.45-2
**Reproducible in staging?:** Y
**Reproducible in production?:** N
**If this was caught during regression testing, add the test name, ID and link from TestRail:** N/A
**Issue reported by:** Applause - Internal Team

**Issue found when executing PR** https://github.com/Expensify/App/pull/49832
## Action Performed:
1. Open a chat
2. Go offline
3. Upload an image
4. Open the image
5. Go online
6. Verify that image preview modal not dismissed

## Expected Result:
Image preview modal is not dismissed after going online, preview of the uploaded offline image is opened

## Actual Result:
Opened preview of the uploaded offline image switches to the preview of previously uploaded image after going online

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [x] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

<details>
  <summary>Add any screenshot/video evidence</summary>

  
</details> 

https://github.com/user-attachments/assets/abcb7911-5a58-4b0f-b770-963ac424beac

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by lanitochka17** [comment]
Production:

https://github.com/user-attachments/assets/fe59703d-ff39-438c-a5d3-0a3310e85759

**Comment by thienlnam** [comment]
This was the note on the linked PR

> There is a known issue in the native platform where the attachment index incorrectly switches even after setting it correctly https://github.com/Expensify/App/pull/49832#issuecomment-2384722144. This is a known upstream issue that should be addressed separately https://github.com/Expensify/App/pull/49832#issuecomment-2385872261.

**Comment by thienlnam** [comment]
cc @s77rt @wildan-m @rlinoz If I'm reading that comment correctly - are we saying that this current bug was broken elsewhere and needs to fixed upstream? Because this is not reproducible on production so it seems like somewhere along the line it was changed

**Comment by AndrewGable** [comment]
Thanks for noticing that @thienlnam - Do we have a follow up issue we can link here or can we just use this as the issue?

**Comment by rlinoz** [comment]
Yeah, it is not reproducible in prod right now because when going online again in prod the modal is dismissed (that is what the PR is fixing)

This are the related upstream issues we found: 
https://github.com/callstack/react-native-pager-view/issues/791
https://github.com/callstack/react-native-pager-view/issues/597

I don't think we have an issue to track that on our side yet

**Comment by s77rt** [comment]
@thienlnam In production I believe the behaviour was that after you get online the modal will be dismissed which is already considered a bug. Now the modal is not dismissed but the attachment changes. Since this was not working at all it is not a regression. The new bug on native is upstream and we should fix it as well however this does not need to be a deploy blocker.

**Comment by thienlnam** [comment]
Ahh okay I see, thanks for the clarification - yeah let's create an issue to track that on our side and then we can solve it seperately

**Comment by AndrewGable** [comment]
Do we need to keep this one open @thienlnam ?

**Comment by thienlnam** [comment]
I imagine this one might get re-reported so was thinking it could be good to leave it open and on hold for the other issue - but feel free to close if you want

**Comment by mallenexpensify** [comment]
We're holding on creating a regression test to [this linked issue](https://github.com/Expensify/App/issues/48173#issuecomment-2407504216), so we'll likely want one here.

**Comment by AndrewGable** [comment]
I'm a bit confused on the next steps here, seems like we are waiting on making a regression test, but we are waiting on this issue to create the regression test?

**Comment by s77rt** [comment]
@AndrewGable Making a regression test now won't make much sense because there is still an ongoing bug and the QA team  won't be able to complete the test. Once https://github.com/Expensify/App/issues/50349 is fixed we can proceed

**Comment by AndrewGable** [comment]
Still on hold

**Comment by AndrewGable** [comment]
Seems like the issue on hold is getting close

**Comment by AndrewGable** [comment]
Deployed to staging, let's wait a little bit longer

**Comment by AndrewGable** [comment]
@s77rt @thienlnam - What's next steps here?

**Comment by s77rt** [comment]
@AndrewGable Let's add the following regression test and close the issue:

1. Open any chat
2. Go offline
3. Upload an image
4. Open the image
5. Go online
6. Verify the image is still displayed and that the modal is not dismissed

**Comment by mallenexpensify** [comment]
Thanks @s77rt test case created, gonna close
- https://github.com/Expensify/Expensify/issues/451331
